src/syscheckd/db/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.12.4)
+
+project(fimdb)
+
+enable_testing()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+get_filename_component(SRC_FOLDER     ${CMAKE_SOURCE_DIR}/../../ ABSOLUTE)
+message("${SRC_FOLDER} ${CMAKE_SOURCE_DIR}")
+if(COVERITY)
+  add_definitions(-D__GNUC__=8)
+endif(COVERITY)
+
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wshadow -Wnon-virtual-dtor -Woverloaded-virtual -Wunused -Wcast-align -Wformat=2 -std=c++14 -pthread")
+
+set(CMAKE_CXX_FLAGS_DEBUG "-g")
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+else()
+  set(CMAKE_CXX_FLAGS_RELEASE "-O3 -s")
+endif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+
+if(FSANITIZE)
+  set(CMAKE_CXX_FLAGS_DEBUG "-g -fsanitize=address,leak,undefined")
+endif(FSANITIZE)
+
+if(NOT LIBOUT)
+set(LIB_DIR ${CMAKE_BINARY_DIR})
+else()
+set(LIB_DIR ${CMAKE_BINARY_DIR}/${LIBOUT})
+endif()
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${LIB_DIR})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${LIB_DIR})
+
+if(APPLE)
+  set(CMAKE_MACOSX_RPATH 1)
+endif(APPLE)
+
+include_directories(${SRC_FOLDER}/)
+include_directories(${SRC_FOLDER}/headers/)
+include_directories(${SRC_FOLDER}/external/sqlite/)
+include_directories(${SRC_FOLDER}/external/nlohmann/)
+include_directories(${SRC_FOLDER}/external/cJSON/)
+include_directories(${SRC_FOLDER}/external/procps/)
+include_directories(${SRC_FOLDER}/external/bzip2/)
+include_directories(${SRC_FOLDER}/external/openssl/include/)
+include_directories(${SRC_FOLDER}/shared_modules/utils)
+include_directories(${SRC_FOLDER}/shared_modules/dbsync/include/)
+include_directories(${SRC_FOLDER}/shared_modules/rsync/include/)
+include_directories(${SRC_FOLDER}/shared_modules/common/)
+include_directories(${SRC_FOLDER}/data_provider/include/)
+include_directories(${SRC_FOLDER}/syscheckd)
+include_directories(${CMAKE_SOURCE_DIR}/include)
+
+link_directories(${SRC_FOLDER}/shared_modules/dbsync/build/lib)
+link_directories(${SRC_FOLDER}/shared_modules/rsync/build/lib)
+link_directories(${SRC_FOLDER}/data_provider/build/lib)
+link_directories(${SRC_FOLDER})
+
+link_directories(${SRC_FOLDER}/external/openssl/)
+link_directories(${SRC_FOLDER}/external/sqlite/)
+link_directories(${SRC_FOLDER}/external/cJSON/)
+link_directories(${SRC_FOLDER}/external/procps/)
+link_directories(${SRC_FOLDER}/external/bzip2/)
```

Comment:
It looks like after the change compilation fails on Linux because the openssl include and link directories have been moved inside a Windows specific block, please review this once more.